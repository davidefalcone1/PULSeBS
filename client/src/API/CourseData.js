//usata per la retrieve della lista dei corsi di un docente
class CourseData{
    constructor(courseId, courseName, teacherId, 
        normalBookingsAvgWeek, cancelledBookingsAvgWeek, waitingBookingsAvgWeek, 
        normalBookingsAvgMonth, cancelledBookingsAvgMonth, waitingBookingsAvgMonth,
        attendanceCountAvgMonth, attendanceCountAvgWeek ) {
        if (courseId) 
            this.courseId = courseId;
        this.courseName = courseName;
        this.teacherId = teacherId;
        this.normalBookingsAvgWeek = normalBookingsAvgWeek;
        this.cancelledBookingsAvgWeek = cancelledBookingsAvgWeek;
        this.waitingBookingsAvgWeek = waitingBookingsAvgWeek;
        this.normalBookingsAvgMonth = normalBookingsAvgMonth;
        this.cancelledBookingsAvgMonth = cancelledBookingsAvgMonth;
        this.waitingBookingsAvgMonth = waitingBookingsAvgMonth;
        this.attendanceCountAvgWeek = attendanceCountAvgWeek;
        this.attendanceCountAvgMonth = attendanceCountAvgMonth;
    }

    static fromJson(json){
        const temp =  Object.assign(new CourseData(), json);
        return temp;
    }
}

export default CourseData;