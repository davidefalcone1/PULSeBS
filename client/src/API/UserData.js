class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

    static fromJson(json){
        const temp =  Object.assign(new UserData(), json);
        return temp;
    }
}

export default UserData;