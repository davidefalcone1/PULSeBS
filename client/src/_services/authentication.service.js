import { BehaviorSubject } from 'rxjs';


const currentUserSubject = new BehaviorSubject(JSON.parse(localStorage.getItem('user')));

export const authenticationService = {
    login,
    logout,
    user: currentUserSubject.asObservable(),
    get currentUserValue () { return currentUserSubject.value }
};

function handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            if ([401, 403].indexOf(response.status) !== -1) {
                // auto logout if 401 Unauthorized or 403 Forbidden response returned from api
                authenticationService.logout();
            }

            const error = (data && data.message) || response.statusText;
            return Promise.reject(error);
        }

        return data;
    });
}

function login(username, password) {
    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ username, password })
    };
    return new Promise(async function (resolve, reject) {
        try {
            const response = await fetch('/users/authenticate', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ username, password }),
            });
            const user = await response.json().then(handleResponse);
            if (response.ok) {
                localStorage.setItem('user', JSON.stringify(user));
                currentUserSubject.next(user);
                resolve(user);
            }
            else
                reject();
        }
        catch (e) {
            reject();
        }
    });

}
function logout() {

    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
    };
    localStorage.removeItem('user');
    currentUserSubject.next(null);
    return fetch('/logout',requestOptions);
}