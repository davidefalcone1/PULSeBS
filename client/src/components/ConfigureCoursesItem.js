import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import Button from 'react-bootstrap/Button';

const configureCourseItem = (props) => {

  return (
    <ListGroup.Item id = {"course-" + props.course.courseId}>
        <div className="d-flex w-100 pt-3 justify-content-between no-gutters">
            <NameField id = {props.course.courseId} name = {props.course.courseName}/>
            <TeacherField id = {props.course.courseId} teacherId = {props.course.teacherId}
                teachersList = {props.teachers}/>
            <ButtonField id = {props.course.courseId} editCourse = {props.editCourse}/>
        </div>
    </ListGroup.Item>
  );
}

function NameField(props){
    return(
        <div className="col-sm-5">
            <h6 id={"nameOfcourse_" + props.id}>
                {props.name}
            </h6>
        </div>
    );
}

function TeacherField(props){
    return(
        <div className="col-sm-5">
            {props.teachersList.map((t) => //per ogni lezione del mio corso
                (t.personId === props.teacherId) &&
                    <h6 id={"teacherOfcourse_" + t.personId + "-" + props.id}>
                        {t.fullName}
                    </h6>
            )}
        </div>
    );
}

function ButtonField(props){
    return(
        <div className="col-sm-2">
            <Button variant="primary" onClick={(event) => {
                event.preventDefault();
                props.editCourse(); //TODO SEND PARAMETERS
            }} id={"buttonFieldOfcourse" + props.id}>
                Edit
            </Button>
        </div>
    );
}

export default configureCourseItem;