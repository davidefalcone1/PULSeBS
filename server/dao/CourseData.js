//usata per la retrieve della lista dei corsi
class CourseData{
    constructor(courseId, courseName, teacherId) {
        if (courseId) 
            this.courseId = courseId;
        this.courseName = courseName;
        this.teacherId = teacherId;
    }
}

module.exports = CourseData;