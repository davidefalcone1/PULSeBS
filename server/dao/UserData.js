class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

}

module.exports = UserData;