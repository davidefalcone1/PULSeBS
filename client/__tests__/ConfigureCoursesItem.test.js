import React from 'react';
import renderer from 'react-test-renderer';
import { AuthContext } from '../src/_services/AuthContext';
import ConfigureCoursesItem from '../src/components/ConfigureCoursesItem';
import { BrowserRouter } from 'react-router-dom';
import CourseData from '../src/API/CourseData';
import UserData from '../src/API/UserData';

describe('ConfigureCoursesItem', ()=>{
    test('Renders correctly', ()=>{
        const component = renderer.create(
            <BrowserRouter>
                <AuthContext.Provider value={{user: new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')}}>
                    <ConfigureCoursesItem course={new CourseData('XY123', 'Software engineering 2', '654321', 2, 3, 5, 6, 7, 3, 5, 7)} teachers={[new UserData(1, '654321', 'Mario Rossi', 'mario.rossi@polito.it')]}/>
                </AuthContext.Provider>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});