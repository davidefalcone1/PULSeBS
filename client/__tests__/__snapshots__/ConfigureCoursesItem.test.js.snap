// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureCoursesItem Renders correctly 1`] = `
<div
  className="list-group-item"
  disabled={false}
  id="course-XY123"
  onClick={[Function]}
>
  <div
    className="d-flex w-100 pt-3 justify-content-between no-gutters"
  >
    <div
      className="col-sm-6"
    >
      <h6
        id="nameOfcourse_XY123"
      >
        Software engineering 2
      </h6>
    </div>
    <div
      className="col-sm-6"
    >
      <h6
        id="teacherOfcourse_654321-XY123"
      >
        Mario Rossi
      </h6>
    </div>
  </div>
</div>
`;
